src/components/pagination/index.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { FC, PropsWithChildren, useMemo } from 'react';
+import { useMultiStyleConfig } from '@chakra-ui/react';
+
+import PaginationButton from './PaginationButton';
+import { ChevronLeftSmallIcon, ChevronRightSmallIcon } from '../icons';
+import Box from '../box';
+
+const Dots = '...';
+const range = (start: number, end: number): Array<number> => {
+  const length = end - start + 1;
+  return Array.from({ length }, (_, idx) => idx + start);
+};
+
+export interface Props {
+  count: number;
+  page: number;
+  siblingCount?: number;
+  onChange: (page: number) => void;
+}
+
+const Pagination: FC<PropsWithChildren<Props>> = (props) => {
+  const { onChange, count, page, siblingCount = 1 } = props;
+  const { paginationContainer, dots } = useMultiStyleConfig('Pagination');
+
+  const paginationRange = useMemo(() => {
+    // Default number of page buttons: firstPage + lastPage + currentPage + left side dots + right side dots
+    const pageButtonsCount = 5;
+    const totalPageButtonsCount = siblingCount + pageButtonsCount;
+
+    if (totalPageButtonsCount >= count) {
+      return range(1, count);
+    }
+
+    const leftSiblingIndex = Math.max(page - siblingCount, 1);
+    const rightSiblingIndex = Math.min(page + siblingCount, count);
+
+    const shouldShowLeftDots = leftSiblingIndex > 2;
+    const shouldShowRightDots = rightSiblingIndex < count - 2;
+
+    const firstPageIndex = 1;
+    const lastPageIndex = count;
+
+    if (!shouldShowLeftDots && shouldShowRightDots) {
```

Comment:
we haven't covered those branches in tests. They are rather complex. Could you try to cover it? thanks!